tool/teleport/common/teleport.go

```diff
@@ -548,11 +549,30 @@ func onConfigDump(flags dumpFlags) error {
 	}
 
 	if configPath != "" {
-		if modules.GetModules().BuildType() == modules.BuildOSS {
-			fmt.Printf("Wrote config to file %q. Now you can start the server. Happy Teleporting!\n", configPath)
+		canWriteToDataDir := utils.CanUserWriteTo(flags.DataDir)
+		canWriteToConfDir := utils.CanUserWriteTo(configPath)
+		requiresRoot := !canWriteToDataDir || !canWriteToConfDir
+
+		fmt.Printf("A Teleport configuration file has been created at %q.\n", configPath)
+		if modules.GetModules().BuildType() != modules.BuildOSS {
+			fmt.Printf("Your license file should be added to %q.\n", flags.LicensePath)
+		}
+		fmt.Printf("To start a teleport node with this configuration file, run:\n\n")
+		if requiresRoot {
+			fmt.Printf("sudo teleport start --config=%q\n\n", configPath)
+			fmt.Printf("Note that starting a Teleport server with this configuration will require root access as:\n")
+			if !canWriteToConfDir {
+				fmt.Printf("- The Teleport configuration is located at %q.\n", configPath)
+			}
+			if !canWriteToDataDir {
+				fmt.Printf("- Teleport will be storing data at %q. To change that, run \"teleport configure\" with the \"--data-dir\" flag.\n", flags.DataDir)
```

Comment:
Nit: If you used apostrophes (`'`) for the string, you wouldn't need to escape quotes.